Sources/Workspace/InitPackage.swift

```diff
@@ -579,12 +579,9 @@ public final class InitPackage {
         }
         try makeDirectories(moduleDir)
 
-        let sourceFileName: String
-        if packageType == .executable {
-            sourceFileName = "main.swift"
-        } else {
-            sourceFileName = "\(typeName).swift"
-        }
+        // If we're creating an executable we can't 't have both a @main delcaration and a Main.swift file.
+        // Handle the edge case of a user creating a project called "Main" by give the generated file a different name.
+        let sourceFileName = (packageType == .executable && typeName == "main") ? "MainEntrypoint.swift" : "\(typeName).swift"
```

Comment:
How about just using uppercase `Main.swift` instead of `MainEntrypoint.swift` if we already uppercase `main` anyway? Don't feel too strongly about it.

Reply:
I was trying to steer people away from having a "Main.swift" in any form. The [guidance on swift.org](https://www.swift.org/getting-started/cli-swiftpm/#a-small-application) states `You can either have a main.swift file or a @main entrypoint, but not both.`, and it would be easy to miss that `main.swift` and `Main.swift` are different.

Also, its my understanding that Windows filenames are typically case insensitive. I haven't tested whether this impacts how the compiler treats main.swift vs Main.swift, but it could factor in.

Reply:
Fair enough!